hud/src/androidLibAndroidTest/kotlin/AndroidHUDTests.kt

```diff
@@ -134,20 +140,35 @@ class AndroidHUDTests {
     }
 
     @Test
-    fun rotateActivity() {
-        val indicator = builder.build {
-            setTitle(LOADING)
-        }.present()
+    fun `hud_should_stay_on_screen_if_it_rotates`() {
+        val indicator = buildAndPresentLoadingHUD()
+
+        rotateScreen()
+
+        // HUD should be on screen
         device.assertTextAppears(LOADING)
         assertTrue(indicator.isVisible)
 
-        // Rotate screen
-        device.setOrientationLeft()
+        device.setOrientationNatural()
+        indicator.dismiss()
+        // Finally should be gone
+        device.assertTextDisappears(LOADING)
+        assertFalse(indicator.isVisible)
+    }
+
+    @Test
+    fun `hud_should_keep_showing_if_app_gos_to_background_and_returns_back`() {
+        val indicator = buildAndPresentLoadingHUD()
+
+        val appLabel = activityRule.activity.localClassName
+        device.pressHome()
+        device.pressRecentApps()
+        device.findObject(UiSelector().text(appLabel)).click()
```

Comment:
This line fails on CI, if you watch the movie there is seems the recent app window is not brought up for some reason. Perhaps you can reproduce using the same emulator as there, but in general I think this method is not super stable.

Reply:
This line simply will not work for each build.. Android doesnt show the app name label in recent apps for all Android versions. If its not displayed, the findObject will fail and the test as well. I guess you could relaunch the activity somehow, but its not gonna be a great test. Lifecycle wise the rotation already has the same behaviour aanyway, so Id just stick with that.